AWS SES -- email notification

vnandwana opened the pull request:
Updated email notification plugin, adding info about using AWS SES.

vnandwana committed:
* Updated email notification plugin, adding info about using AWS SES.

sbrossie commented on userguide/tutorials/metrics-cloudwatch.adoc:
```diff
@@ -60,8 +60,20 @@ AWS X-Ray is a service that collects and analyzes data from your applications. A
-curl https://s3.dualstack.us-east-1.amazonaws.com/aws-xray-assets.us-east-1/xray-daemon/aws-xray-daemon-3.x.deb -o /home/ubuntu/xray.deb
-sudo dpkg -i /home/ubuntu/xray.deb
+# Download the X-Ray daemon package
+wget https://s3.us-east-2.amazonaws.com/aws-xray-assets.us-east-2/xray-daemon/aws-xray-daemon-3.x.deb
```
Would it make sense to include this in our AMI - I thought we had discussed this already?

vnandwana replied:
Sorry about this -- I created this branch based on v3 which already had these XRay changes. There was already a PR for these XRay changes, to avoid confusion I'm closing this old PR -- https://github.com/killbill/killbill-docs/pull/540.

sbrossie reviewed (commented):
Is there anyway we can make it easier for the users - not have to install packages, ... ?

sbrossie commented on userguide/tutorials/email-notification-plugin.adoc:
```diff
@@ -61,12 +63,36 @@ The email-notification-plugin requires some additional database tables. To creat
 
 [[tenant-config]]
-. *Configure the tenant* using the following cURL command (Replace `<events>` with a comma-separated list of the events for which the emails need to be sent. For example, to configure the tenant to send emails for *INVOICE_CREATION* and *INVOICE_PAYMENT_SUCCESS*, specify `org.killbill.billing.plugin.email-notifications.defaultEvents=INVOICE_CREATION,INVOICE_PAYMENT_SUCCESS`):
+*Configure the tenant* using the following cURL command (Replace `<events>` with a comma-separated list of the events for which the emails need to be sent. For example, to configure the tenant to send emails for *INVOICE_CREATION* and *INVOICE_PAYMENT_SUCCESS*, specify `org.killbill.billing.plugin.email-notifications.defaultEvents=INVOICE_CREATION,INVOICE_PAYMENT_SUCCESS`):
```
Is this on purpose (no more space) ?

vnandwana replied:
This is how it was earlier. I just removed the dot (.) from the beginning of the line so that it doesn't add a number section.

reshmabidikar replied:
Is there a specific reason to get rid of the numbering?

reshmabidikar replied:
I think tenant/account configuration (Steps 1 and 2 in the [plugin configuration section](https://docs.killbill.io/latest/email-notification-plugin#plugin_configuration)) would be required when AWS SES is used as well. 

If you agree, we can maybe add a section below [SMTP Server Notes](https://docs.killbill.io/latest/email-notification-plugin#smtp_server_notes) called `AWS SES Specific Configuration` that only includes the additional configuration for AWS SES and make it clear that this is only the additional configuration.

vnandwana replied:
The tenant configuration mentioned in the SMTP section is not required for AWS SES. I have made some changes to the doc, hopefully, these comments are addressed. Kindly check when you get a chance. Thank you.

sbrossie commented on userguide/tutorials/email-notification-plugin.adoc:
```diff
@@ -85,11 +111,13 @@ org.killbill.billing.plugin.email-notifications.smtp.useSSL=false
+
+
+=== Configure Events
+*Configure the account* using the following `cURL` command (Replace `{accountId}` with the id of the account for which emails need to be sent and `<events>` with a comma-separated list of the events for which the emails need to be sent. For example, to configure the account to send emails for *INVOICE_CREATION* and *INVOICE_PAYMENT_SUCCESS*, specify `"INVOICE_CREATION", "INVOICE_PAYMENT_SUCCESS"` ...
```
Same here: Is this on purpose (no more space) ?

vnandwana replied:
here as well, I just removed the dot (.) from the beginning of the line so that it doesn't add a number section.

reshmabidikar replied:
Is there a specific reason to get rid of the numbering?

sbrossie reviewed (commented):
Left a few comments to address.

vnandwana committed:
* XRay daemon now comes preinstalled with AMI. Removed the installation step.

vnandwana commented:
> Is there anyway we can make it easier for the users - not have to install packages, ... ?

Could you please elaborate on your thoughts? A user needs to install this plugin (just like any other plugins), are you referring to simplifying the AWS SES configuration process -- https://github.com/killbill/killbill-docs/pull/550/files#diff-371363a9cb5621593e52291217cd15e1d82ba9479fb62d0bfcc3e4ce6072fb78R69

vnandwana commented:
> > Is there anyway we can make it easier for the users - not have to install packages, ... ?
> 
> Could you please elaborate on your thoughts? A user needs to install this plugin (just like any other plugins), are you referring to simplifying the AWS SES configuration process -- https://github.com/killbill/killbill-docs/pull/550/files#diff-371363a9cb5621593e52291217cd15e1d82ba9479fb62d0bfcc3e4ce6072fb78R69

This PR had X-Ray daemon installation as well, if you're referring to simplifying this, its already been done.

vnandwana committed:
Addressed review comments.

vnandwana commented:
@reshmabidikar @sbrossie I have addressed the review comments, kindly check at your convenience. I have verified these changes in preview mode using the AsciiDoc plugin for IntelliJ IDEA.

Please let me know if any further changes are required. Thank you.

sbrossie committed:
Merge branch 'v3' into ses-email-notification

sbrossie commented:
@vnandwana I fixed the merge conflict. @reshmabidikar Can you double check your comments have been addressed - and if so, move ahead with merging this PR?

reshmabidikar commented on userguide/tutorials/email-notification-plugin.adoc:
```diff
@@ -1,13 +1,15 @@
 
 * Ensure that you have https://curl.haxx.se/[_cURL_] installed. If you are on Windows, we recommend that you use https://git-scm.com/download/win[_Git Bash_] to run the `cURL` commands.
 
+* If you plan to use AWS SES, obtain your AWS SES credentials (access key and secret key) and ensure that the IAM user has the ses:SendEmail permission.
```
```suggestion
* If you plan to use AWS SES, obtain your AWS SES credentials (access key and secret key) and ensure that the IAM user has the `ses:SendEmail permission`.
```

reshmabidikar commented on userguide/tutorials/email-notification-plugin.adoc:
```diff
@@ -85,29 +111,24 @@ org.killbill.billing.plugin.email-notifications.smtp.useSSL=false
 
-.. If a tenant is configured with some events, but the account is not configured, then emails will be sent based on what is configured at the tenant level.
+[[smtp_server_notes]]
+== SMTP Server Notes
```
This should be `H3` so that the doc reads better:
```suggestion
=== SMTP Server Notes
```

reshmabidikar commented on userguide/tutorials/email-notification-plugin.adoc:
```diff
@@ -61,12 +63,36 @@ The email-notification-plugin requires some additional database tables. To creat
+=== SMTP Configuration
+A tenant needs to be configured with the SMTP properties required for sending emails. Additionally, the tenant can also be configured with the events for which emails should be sent. In addition to the per-tenant configuration, the plugin also allows a more granular account-level configuration for the set of emails to be sent for the particular account. Thus, either the tenant or the account need ...
 
 The tenant/account configuration can be done by executing the required API endpoints via *cURL* as explained below:
```
```suggestion
The tenant configuration can be done by executing the required API endpoints via *cURL* as explained below:
```

reshmabidikar commented on userguide/tutorials/email-notification-plugin.adoc:
```diff
@@ -61,12 +63,36 @@ The email-notification-plugin requires some additional database tables. To creat
+*Note:* Ensure that the credentials used have the necessary permissions, specifically `ses:SendEmail`, on the resource arn:aws:ses:<region>:<account>:identity/<sender's email>.
+
+=== SMTP Configuration
+A tenant needs to be configured with the SMTP properties required for sending emails. Additionally, the tenant can also be configured with the events for which emails should be sent. In addition to the per-tenant configuration, the plugin also allows a more granular account-level configuration for the set of emails to be sent for the particular account. Thus, either the tenant or the account need ...
```
Move the text `In addition to the per-tenant configuration, the plugin also allows a more granular account-level configuration for the set of emails to be sent for the particular account. Thus, either the tenant or the account needs to be configured with the events for which the email needs to be sent.` to the `Configure Events` section below.

vnandwana replied:
Didn't move this as the statement is good for SMTP configuration.

reshmabidikar replied:
I agree, but from what I read, the account configuration is now moved to the `Configure Events` section, hence I suggested moving the above text to the `Configure Events`. If you disagree, feel free to leave it as it is and merge the PR.

vnandwana replied:
Lets keep this in SMTP as the following is not applicable for SES:
 "per-tenant configuration" or "either the tenant or the account needs to be configured".
 
I'm not authorized to merge code to any of the public repositories.

vnandwana replied:
@reshmabidikar If no further changes are required, kindly merge this branch. I don't have permission to merge code to this branch. Thank you.

reshmabidikar commented on userguide/tutorials/email-notification-plugin.adoc:
```diff
@@ -61,12 +63,36 @@ The email-notification-plugin requires some additional database tables. To creat
+org.killbill.mail.from=<Default sender's email>
+org.killbill.aws.region=<AWS region, default region is us-east-1>
+
+AWS SES requires authentication for sending emails. System-wide environment variables can be configured for all users by appending them to the /etc/environment file. Open the `/etc/environment` file in a text editor with root privileges, set the following variables:
```
```suggestion
AWS SES requires authentication for sending emails. System-wide environment variables can be configured for all users by appending them to the `/etc/environment` file. Open the `/etc/environment` file in a text editor with root privileges, set the following variables:
```

reshmabidikar reviewed (commented):
Looks good Vijay. I reviewed the doc from [your fork ](https://github.com/vnandwana/killbill-docs/blob/ses-email-notification/userguide/tutorials/email-notification-plugin.adoc) to see how it reads and have left a few minor comments to address.

vnandwana committed:
Addressed review comments.

vnandwana committed:
Addressed review comments.

vnandwana commented:
@reshmabidikar @sbrossie Thank you for your review comments. I have addressed all these comments. Could you take a one final look?

reshmabidikar reviewed (approved):
Apart from https://github.com/killbill/killbill-docs/pull/550#discussion_r1624514495, everything looks good. If you think the change in my comment is not required, feel free to merge and we can revisit once PR is merged.
